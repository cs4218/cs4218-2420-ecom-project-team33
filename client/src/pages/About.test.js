import React from "react";
import { render, screen } from "@testing-library/react";
import About from "./About";
import "@testing-library/jest-dom";

jest.mock("./../components/Layout", () =>
  jest.fn(({ children }) => <div>{children}</div>)
);

describe("About Component", () => {
  
  it("displays the about image with correct attributes", () => {
    render(<About />);
    
    const aboutImage = screen.getByAltText("contactus");
    expect(aboutImage).toBeInTheDocument();
    expect(aboutImage).toHaveAttribute("src", "/images/about.jpeg");
    expect(aboutImage).toHaveStyle("width: 100%");
  });

  it("contains the text content section", () => {
    render(<About />);    
    // Check for the paragraph
    const paragraph = screen.getByText("Add text");
    expect(paragraph).toBeInTheDocument();
  });

  it("passes the correct title to Layout component", () => {
    render(<About />);
    
    // Check if Layout was called with the correct title prop
    expect(jest.mocked(require("./../components/Layout")).mock.calls[0][0].title).toBe("About us - Ecommerce app");
  });

  it("matches snapshot", () => {
    const { container } = render(<About />);
    expect(container).toMatchSnapshot();
  });
});
