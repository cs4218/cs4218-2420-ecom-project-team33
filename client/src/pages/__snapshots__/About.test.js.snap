// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="row contactus "
    >
      <div
        class="col-md-6 "
      >
        <img
          alt="contactus"
          src="/images/about.jpeg"
          style="width: 100%;"
        />
      </div>
      <div
        class="col-md-4"
      >
        <p
          class="text-justify mt-2"
        >
          Add text
        </p>
      </div>
    </div>
  </div>
</div>
`;
